### ISSUE TITLE
Cannot require sqlite3 in both a main thread and a worker thread

### ISSUE THREAD

**Issue by mohd-akram** [mapbox/node-sqlite3#1223]
OS: macOS 10.14.6
`node` version: `12.10.0`
`sqlite3` version:  `4.1.0`

1. Create `main.js`:

```javascript
const { Worker } = require('worker_threads');
require('sqlite3');
new Worker('./worker.js');
```

2. Create `worker.js`:

```javascript
require('sqlite3');
```

3. Run `node main.js`

Output:

```

events.js:186
      throw er; // Unhandled 'error' event
      ^
Error: Module did not self-register.
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:977:18)
    at Module.load (internal/modules/cjs/loader.js:790:32)
    at Function.Module._load (internal/modules/cjs/loader.js:703:12)
    at Module.require (internal/modules/cjs/loader.js:830:19)
    at require (internal/modules/cjs/helpers.js:68:18)
    at Object.<anonymous> (/Users/user/Downloads/test-sqlite3/node_modules/sqlite3/lib/sqlite3.js:4:15)
    at Module._compile (internal/modules/cjs/loader.js:936:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:947:10)
    at Module.load (internal/modules/cjs/loader.js:790:32)
    at Function.Module._load (internal/modules/cjs/loader.js:703:12)
Emitted 'error' event on Worker instance at:
    at Worker.[kOnErrorMessage] (internal/worker.js:176:10)
    at Worker.[kOnMessage] (internal/worker.js:186:37)
    at MessagePort.<anonymous> (internal/worker.js:118:57)
    at MessagePort.emit (events.js:209:13)
    at MessagePort.onmessage (internal/worker/io.js:70:8)
```

**Comment by kewde** [comment]
Hi,

worker_thread is unsupported at the moment.
See https://github.com/mapbox/node-sqlite3/pull/1007

and the PR was reverted:
https://github.com/mapbox/node-sqlite3/pull/1009

I don't know why it was reverted, so I will investigate.

**Comment by pacop** [comment]
Hi, 

is some news right here? Thanks @kewde

**Comment by mohd-akram** [comment]
This is fixed in v5.0.0, but there are other issues with worker threads.

**Comment by pnsvk** [comment]
@mohd-akram Can you please point to the other issues with worker threads ?
